backend/src/main/java/com/bakdata/conquery/models/index/MapInternToExternMapper.java

```diff
@@ -98,24 +94,52 @@ public synchronized void init() {
 		dataset = storage.getDataset().getId();
 
 		final URI resolvedURI = FileUtil.getResolvedUri(config.getIndex().getBaseUrl(), csv);
-		log.trace("Resolved mapping reference csv url '{}': {}", this.getId(), resolvedURI);
+		log.trace("Resolved mapping reference csv url '{}': {}", getId(), resolvedURI);
 
-		MapIndexKey key = new MapIndexKey(resolvedURI, internalColumn, externalTemplate);
+		final IndexKey<String> key = new MapIndexKey(resolvedURI, internalColumn, externalTemplate, allowMultiple);
 
 		int2ext = CompletableFuture.supplyAsync(() -> {
+									   try {
+										   return mapIndex.<Index<String>, String>getIndex(key);
+									   }
+									   catch (IndexCreationException e) {
+										   throw new IllegalStateException(e);
+									   }
+								   })
+								   .whenComplete((m, e) -> {
+									   if (e != null) {
+										   log.warn("Unable to get index: {} (enable TRACE for exception)", key, log.isTraceEnabled() ? e : null);
+									   }
+								   });
+	}
+
+	@Override
+	public Collection<String> externalMultiple(String internalValue) {
+		if (indexAvailable()) {
 			try {
-				return mapIndex.getIndex(key);
-			}
-			catch (IndexCreationException e) {
-				throw new IllegalStateException(e);
+				return int2ext.get().externalMultiple(internalValue, internalValue);
 			}
-		}).whenComplete((m, e) -> {
-			if (e != null) {
-				log.warn("Unable to get index: {} (enable TRACE for exception)", key, (Exception) (log.isTraceEnabled() ? e : null));
+			catch (InterruptedException | ExecutionException e) {
+				// Should never be reached
+				log.warn("Unable to resolve mapping for internal value {} (enable TRACE for exception)", internalValue, log.isTraceEnabled() ? e : null);
 			}
-		});
+		}
+
+		return List.of(internalValue);
 	}
 
+	private boolean indexAvailable() {
+		if (!initialized()) {
+			log.trace("Skip mapping {}, because mapper is not yet initialized", getId());
```

Comment:
Aber das mapping wird doch geskipped?

Reply:
Kommt auf die Callsite an, was sie mit dem Aufruf erreichen wollte. Ich würde die logmessage anpassen
```suggestion
			log.trace("Mapping {} index not available, because mapper is not yet initialized", getId());
```